File: 647 Palindromic Substrings/Palindromic Substrings.js
/**
 * @param {string} s
 * @return {number}
 */
const countSubstrings = (s) => {
  let count = 0;
  for (let i = 0; i < s.length; i++) {
    let left = (right = i);

    while (left >= 0 && right < s.length && s[left] === s[right]) {
      count++;
      left--;
      right++;
    }

    left = i;
    right = i + 1;

    while (left >= 0 && right < s.length && s[left] === s[right]) {
      count++;
      left--;
      right++;
    }
  }

  return count;
};

File: 647 Palindromic Substrings/Palindromic Substrings.py
#Python3 versiyonunu seçmeyi unutma =)
class Solution:
    def countSubstrings(self, s: str) -> int:
        count =0
        for i in range(0,len(s)):
            left = right =i

            while left >=0 and right < len(s) and s[left] == s[right] :
                count +=1
                left -=1
                right +=1

            left = i
            right = i+1

            while left >=0 and right < len(s) and s[left] == s[right] :
                count +=1
                left -=1
                right +=1
        return count